core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -227,6 +227,7 @@
 import static org.apache.calcite.sql.fun.SqlLibraryOperators.SINH;
 import static org.apache.calcite.sql.fun.SqlLibraryOperators.SORT_ARRAY;
 import static org.apache.calcite.sql.fun.SqlLibraryOperators.SOUNDEX;
+import static org.apache.calcite.sql.fun.SqlLibraryOperators.SOUNDEX2;
```

Comment:
I feel like SOUNDEX2 implies it takes 2 arguments, maybe SOUNDEX_SPARK would be a better name? Similar to the SUBSTR or FLOOR/CEIL functions.

Reply:
Maybe the behavior is still a bit different, such as calcite mysql library(it limits 4 chars) now
```
mysql> SELECT SOUNDEX('Quadratically');
         -> 'Q363'
```

And mysql:
```
mysql> SELECT SOUNDEX('Quadratically');
         -> 'Q36324'
```

But this behavior is also different from spark, so we may need to deal with each dialect separately later. I will create an issue describing the syntax difference of this function in mysql when i'm free. WDYT?